src/parser/shared/modules/HotTracker.ts

```diff
@@ -530,21 +540,74 @@ class HotTracker extends Analyzer {
+   */
+  abstract _generateHotList(): Spell[];
+}
+
+export type TrackersByPlayerAndSpell = { [key: number]: TrackersBySpell };
+
+export type TrackersBySpell = { [key: number]: Tracker };
+
+// a tracking object for a specific instance of the player's HoT on a target
+export interface Tracker {
+  start: number; // timestamp when the tracked HoT was applied
+  end: number; // timestamp when the tracked HoT is projected to end (dynamically updated by extensions)
+  originalEnd: number; // timestamp when the tracked HoT was originally projected to end (before extensions)
+  spellId: number; // the HoT's spellId
+  name: string; // the HoT's name, for logging purposes
+  ticks: Tick[]; // listing of ticks recorded by this HoT
+  attributions: Attribution[]; // listing of attributions attached to this HoT
+  extensions: Extension[]; // listing of extensions attached to this HoT
+  boosts: Boost[]; // listing of boosts attached to this HoT
+  healingAfterOriginalEnd: number; // healing that occured after the original end (can be attributed to extensions)
+  maxDuration: number; // the 'true max duration' that extensions will obey (see Rising Mist)
+}
+
+// a record of a HoT's tick
+export interface Tick {
+  healing: number; // the tick's effective healing
+  timestamp: number; // the tick's timestamp
+}
+
+// a record of healing attributable to an effect
+export interface Attribution {
+  attributionId: number | null; // the ID of the effect attributed, or null in the case of a hardcast
+  name: string; // the name of the attribution, for logging purposes only
+  healing: number; // the amount of effective healing attributable - will be updated
+  procs: number; // the number of times this attribution was made - will be updated
+  totalExtension?: number; // the number of total milliseconds extended (for an extension attribution)
+}
+
+// a record of an extension to a HoT
+export interface Extension {
+  amount: number; // the length of the extension in ms
+  attribution: Attribution; // the attribution of this extension
+}
+
+// a record of a boost to a HoT (for its full duration)
+export interface Boost {
+  increase: number; // the amount of the healing increase - ex. 0.15 means a 15% increase
+  attribution: Attribution; // the attribution of this boost
+}
+
+// A mapping from spell ID to that spell's HotInfo
+export type HotInfoMap = { [key: number]: HotInfo };
+
+// information about a Heal over Time spell specific to tracking
+export interface HotInfo {
+  duration: number; //HoT's base duration, in ms
+  tickPeriod: number; //HoT's base period between ticks, in ms
+  maxDuration?: (e: ApplyBuffEvent, c: Combatant) => number; // TODO used by MW - description?
+  bouncy?: boolean; // true iff a hot will bounce to another target on any event
+  durationConditions?: (e: ApplyBuffEvent | RefreshBuffEvent, c: Combatant) => number; // TODO used by MW - description?
```

Comment:
I think Abelito was the one who added this functionality - not super clear on why it's needed or how it works, but I tried to leave it untouched. I confirmed that the Mistweaver module still returns the same results afterwards.

Reply:
I'm not working on this project anymore but you can look at HotTrackerMW.tsx for an example of durationConditions. Not all hots have a 1 constant duration. Thunder focus Tea increases Renewing Mists duration by 10 seconds.
Then Rising mist has a limit of only being able to double its init duration so TFT -> rem would have a max of 60 seconds compared to the normal maximum of 40. 
So this was added to the core incase blizzard ever uses it again or if you want to do pandemic checking or something generically

Reply:
As a note. Combatant needs to be passed in or it will not be able to find it in the function its calling.

Reply:
Thanks for giving the context. I had mostly figured this out but wasn't confident enough to write the definitive docs for it. I'll update the docs with your info.

And yeah, I discovered the thing with Combatant - but I think it was harmless to make Combatant a separate field rather than adding it to the Event. TypeScript gets upset when you try to tack things to defined objects.